db/ops.go

```diff
@@ -32,7 +32,7 @@ func (l *Loader) Insert(tableName string, primaryKey map[string]string, data map
 		l.entries.Set(tableName, entry)
 	}
 
-	if _, found := entry.Get(uniqueID); found {
+	if _, found := entry.Get(uniqueID); found && !l.getDialect().AllowPkDuplicates() {
```

Comment:
Is it something that should be controllable of do you think the default to allow duplicate inserts is finy for everyone?

Reply:
I think this should be the default, imo this check gives a false impression about properly handling duplicates but fails to do so on re-running ranges or inaccurate cursors due to the missing multi table transactions. Handling duplicates in Clickhouse should either be done using Replacing engines or externally (for example using Kafka).